src/main/java/org/fundacionjala/coding/danielM/bank_ocr/BankOcr.java

```diff
@@ -0,0 +1,170 @@
+    private static final int CORRECT_ACCOUNT_LENGTH = 9;
+    private static final int MULTIPLY_FACTOR = 9;
+
+    /**
+     * Constructor private.
+     */
+    public BankOcr() {
+        stringMap.put(KEY.ZERO.ordinal(),
+                          " _ "
+                        + "| |"
+                        + "|_|");
+        stringMap.put(KEY.ONE.ordinal(),
+                          "  |"
+                        + "  |"
+                        + "  |");
+        stringMap.put(KEY.TWO.ordinal(),
+                          " _ "
+                        + " _|"
+                        + "|_ ");
+        stringMap.put(KEY.THREE.ordinal(),
+                          "__ "
+                        + " _|"
+                        + "__|");
+        stringMap.put(KEY.FOUR.ordinal(),
+                          "   "
+                        + "|_|"
+                        + "  |");
+        stringMap.put(KEY.FIVE.ordinal(),
+                          " __"
+                        + "|__"
+                        + " __|");
+        stringMap.put(KEY.SIX.ordinal(),
+                          " __"
+                        + "|__"
+                        + "|__|");
+        stringMap.put(KEY.SEVEN.ordinal(),
+                          "__ "
+                        + "  |"
+                        + "  |");
+        stringMap.put(KEY.EIGHT.ordinal(),
+                          " _ "
+                        + "|_|"
+                        + "|_|");
+        stringMap.put(KEY.NINE.ordinal(),
+                          " _ "
+                        + "|_|"
+                        + " _|");
+    }
+
+    /**
+     * @param value used to obtain its int representation
+     * @return "?" if the value is not in range 0-9 otherwise
+     * return a number between 0 and 9.
+     */
+    private String getKey(final String value) {
+        String key = "?";
+        for (Map.Entry<Integer, String> entry : stringMap.entrySet()) {
+            if (entry.getValue().equalsIgnoreCase(value)) {
+                key = entry.getKey().toString();
+            }
+        }
+        return key;
+    }
+
+    /**
+     * This method determines if an scanned account is legible.
+     *
+     * @param account image of type String.
+     * @return true or false
+     */
+    private boolean isLegible(final String account) {
+        boolean isCorrect = true;
+
+        for (int i = 0; i < account.length(); i++) {
+            if (!Character.isDigit(account.charAt(i))) {
+                isCorrect = false;
+                break;
+            }
+        }
+        return isCorrect;
+    }
+
+    /**
+     * This method validate if an account is correct or not.
+     *
+     * @param account String variable is the account to validate
+     * @return true if the passed account is true, false otherwise.
+     */
+    public boolean validateAccount(final String account) {
+        if (account.length() == CORRECT_ACCOUNT_LENGTH && isLegible(account)) {
```

Comment:
you can improve this method.
instead of using if else condition.

```java
public boolean validateAccount(final String account) {
        if (account.length() == CORRECT_ACCOUNT_LENGTH && isLegible(account)) {
            return false;
        }

        String[] acct = account.split("");
        int checksum = 0;
        int factor = MULTIPLY_FACTOR;
        for (String value : acct) {
            checksum += Integer.parseInt(value) * factor;
            factor--;
        }

        return checksum % MODULUS_FACTOR == 0;
    }
```